First run fails with missing .pid file

When running it for the first time, the process panics because the `daemon.pid` file doesn't exist. The workaround is simple but this needs to be fixed.

Workaround:
```bash
mkdir -p ~/.go-http-cli
echo 10 > ~/.go-http-cli/daemon.pid
```

This is the error:
```
panic: open /home/appuser/.go-http-cli/daemon.pid: no such file or directory

goroutine 1 [running]:
github.com/visola/go-http-cli/daemon.KillDaemon()
	/Users/vinicius.isola/git/go-workspace/src/github.com/visola/go-http-cli/daemon/process.go:48 +0x89
github.com/visola/go-http-cli/daemon.EnsureDaemon(0x16, 0xc42006ae40)
	/Users/vinicius.isola/git/go-workspace/src/github.com/visola/go-http-cli/daemon/process.go:30 +0x59
main.main()
	/Users/vinicius.isola/git/go-workspace/src/github.com/visola/go-http-cli/binaries/http/main.go:15 +0x37
```

--- a/daemon/process.go
+++ b/daemon/process.go
@@ -44,6 +44,11 @@ func EnsureDaemon() error {
 // KillDaemon will kill the process with the PID stored in the daemon.pid file
 func KillDaemon() {
 	pid, pidError := getDaemonPID()
+
+	if os.IsNotExist(pidError) {
+		return
+	}
+
 	if pidError != nil {
 		panic(pidError)
 	}
@@ -58,7 +63,7 @@ func KillDaemon() {
 
 // WriteDaemonPID writes the PID of the current process to a file in the go-http-cli process dir
 func WriteDaemonPID() error {
-	processDir, dirError := getProcessDirectory()
+	processDir, dirError := ensureProcessDirectory()
 	if dirError != nil {
 		return dirError
 	}
@@ -78,7 +83,7 @@ func WriteDaemonPID() error {
 }
 
 func getDaemonPID() (int, error) {
-	processDir, dirError := getProcessDirectory()
+	processDir, dirError := ensureProcessDirectory()
 	if dirError != nil {
 		return 0, dirError
 	}
@@ -91,16 +96,22 @@ func getDaemonPID() (int, error) {
 	return strconv.Atoi(strings.TrimSpace(string(buffer)))
 }
 
-func getProcessDirectory() (string, error) {
+func ensureProcessDirectory() (string, error) {
 	user, err := user.Current()
 	if err != nil {
 		return "", err
 	}
-	return user.HomeDir + goHTTPCLIDirectory, nil
+
+	processDirPath := user.HomeDir + goHTTPCLIDirectory
+	if _, err := os.Stat(processDirPath); os.IsNotExist(err) {
+		os.Mkdir(processDirPath, 0700)
+	}
+
+	return processDirPath, nil
 }
 
 func startDaemon() {
-	processDir, dirError := getProcessDirectory()
+	processDir, dirError := ensureProcessDirectory()
 	if dirError != nil {
 		panic(dirError)
 	}

